### PR TITLE
82 handle misc dotfiles

### PR DESCRIPTION
**Pull request by joe-wolfe21**
closes: #82 

---

## What

As reported in the bug, if misc files sneak into the `root_dir`, the library and UI return empty data rather than just skipping the misc file. This was noticed by `.ipynb_checkpoints` folders from experimentation in Lab.

This handles misc files under the `root_dir` for both the sync library.

## How to Test

Verify tests included handle edge cases and the bug as described in #82 is no longer an issue

### CODE REVIEW

File: rubicon/repository/base.py [hunk 1]
```diff
@@ -38,12 +38,29 @@ def __init__(self, root_dir, **storage_options):
...
     def _ls_directories_only(self, path):
         """Returns the names of all the directories at path `path`."""
-        return [
+        directories = [
             os.path.join(p.get("name"), "metadata.json")
             for p in self.filesystem.ls(path, detail=True)
             if p.get("type", p.get("StorageClass")).lower() == "directory"
         ]
 
+        # TODO - discuss changing this to ignore directories that have dotfiles in their paths
```
Comment (joe-wolfe21) [inline rubicon/repository/base.py:47]: @ryanSoley pointed this out -- leaving this up for discussion. There are pros and cons to both. One pro I like of relying on `_cat_paths` is that it's extensible to other errors we may encounter in the future rather than only focusing on the dotfile problem

Comment (ryanSoley) [reply rubicon/repository/base.py:47]: I'm cool with the `_cat_paths` function. that makes sense to me. it may be useful to throw a warning when it hits an error in there

**Review by ryanSoley — APPROVED** [review]
this is great! super glad this won't bug anyone anymore
